source/slang/hlsl.meta.slang

```diff
@@ -2822,82 +2967,82 @@ __target_intrinsic(hlsl)
 __glsl_extension(GL_KHR_shader_subgroup_arithmetic)
 __spirv_version(1.3)
 //__target_intrinsic(glsl, "subgroupExclusiveAnd($0)")
-__target_intrinsic(cuda, "_wavePrefixAnd($0, _getMultiPrefixMask(($1).x))")
+__target_intrinsic(cuda, "_wavePrefixAnd(_getMultiPrefixMask(($1).x), $0)")
```

Comment:
Having to switch the order of operands here is kind of clumsy, and seems to argue in favor of just going with the HLSL `WaveMultiPrefix` convention everywhere, even if it isn't the one we might have chosen ourselves.

Reply:
Ideally, once `WaveMask` is its own type, we'd have overloads of these functions that take the current `uint4`, as well as an overload that takes a `WaveMask`. The `uint4` version could then forward to the `WaveMask` version, by assuming that we have a `uint4->WaveMask` intrinsic conversion, which would compile to an identity op for HLSL, and to a `.x` projection for CUDA.

It seems like some careful cleanup over time could move some of the work around defining these ops into the stdlib and thus save us some of the complexity in these `__target_intrinsic` lines.